doc: clarify docs fs.watch exception may be emitted (#32513)
nodejs/node: doc/api/fs.md

* On Aix systems, this feature depends on [`AHAFS`][], which must be enabled.

If the underlying functionality is not available for some reason, then

> `fs.watch` will not be able to function (an exception may be thrown).

Reviewer: Which is it? It will not function and will throw an exception? Or it may not function and may throw an exception? Either way, 

```suggestion
`fs.watch` will not be able to function and may throw an exception.
```

> `fs.watch` will not be able to function (may an exception will be thrown).

Reviewer 2: ```suggestion
`fs.watch` will not be able to function (an exception may be thrown).
```

Author: Thank you, fixed!

For example, watching files or directories can be unreliable, and in some

> cases impossible, on network file systems (NFS, SMB, etc), or host file systems

Reviewer: ```suggestion
cases impossible, on network file systems (NFS, SMB, etc) or host file systems
```

> when using virtualization software such as Vagrant, Docker, etc.

Reviewer: ```suggestion
when using virtualization software such as Vagrant or Docker.
```

Author: Thanks for reviewing! it is fixed now in 9c5e8fa
